Importar librerías necesarias

In [1]:
import pandas as pd

In [2]:
df_airbnb = pd.read_csv('/Users/melaniealvarez/Documents/Octavo semestres/Data Mining/Pset2/data/raw/Airbnb_Data.csv')
print(df_airbnb.head())

         id  log_price property_type        room_type  \
0   6901257   5.010635     Apartment  Entire home/apt   
1   6304928   5.129899     Apartment  Entire home/apt   
2   7919400   4.976734     Apartment  Entire home/apt   
3  13418779   6.620073         House  Entire home/apt   
4   3808709   4.744932     Apartment  Entire home/apt   

                                           amenities  accommodates  bathrooms  \
0  {"Wireless Internet","Air conditioning",Kitche...             3        1.0   
1  {"Wireless Internet","Air conditioning",Kitche...             7        1.0   
2  {TV,"Cable TV","Wireless Internet","Air condit...             5        1.0   
3  {TV,"Cable TV",Internet,"Wireless Internet",Ki...             4        1.0   
4  {TV,Internet,"Wireless Internet","Air conditio...             2        1.0   

   bed_type cancellation_policy  cleaning_fee  ...   latitude   longitude  \
0  Real Bed              strict          True  ...  40.696524  -73.991617   
1  Real Bed     

In [3]:
df_airbnb.shape

(74111, 29)

In [4]:
df_airbnb.columns

Index(['id', 'log_price', 'property_type', 'room_type', 'amenities',
       'accommodates', 'bathrooms', 'bed_type', 'cancellation_policy',
       'cleaning_fee', 'city', 'description', 'first_review',
       'host_has_profile_pic', 'host_identity_verified', 'host_response_rate',
       'host_since', 'instant_bookable', 'last_review', 'latitude',
       'longitude', 'name', 'neighbourhood', 'number_of_reviews',
       'review_scores_rating', 'thumbnail_url', 'zipcode', 'bedrooms', 'beds'],
      dtype='object')

In [5]:
df_airbnb.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 74111 entries, 0 to 74110
Data columns (total 29 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   id                      74111 non-null  int64  
 1   log_price               74111 non-null  float64
 2   property_type           74111 non-null  object 
 3   room_type               74111 non-null  object 
 4   amenities               74111 non-null  object 
 5   accommodates            74111 non-null  int64  
 6   bathrooms               73911 non-null  float64
 7   bed_type                74111 non-null  object 
 8   cancellation_policy     74111 non-null  object 
 9   cleaning_fee            74111 non-null  bool   
 10  city                    74111 non-null  object 
 11  description             74111 non-null  object 
 12  first_review            58247 non-null  object 
 13  host_has_profile_pic    73923 non-null  object 
 14  host_identity_verified  73923 non-null

POsibles variables a considerar para el modelo

 0   id                      74111 non-null  int64  
 1   log_price               74111 non-null  float64
 2   property_type           74111 non-null  object 
 3   room_type               74111 non-null  object 
 4   amenities               74111 non-null  object 
 5   accommodates            74111 non-null  int64  
 9   cleaning_fee            74111 non-null  bool   
 10  city                    74111 non-null  object 
 24  review_scores_rating    57389 non-null  float64
 26  zipcode                 73145 non-null  object 
 27  bedrooms                74020 non-null  float64

In [6]:
df_airbnb['id'].duplicated().sum() # total de duplicados en la columna id

0

In [7]:
df_airbnb['bedrooms'].isna().sum()

91

In [8]:
df_airbnb[['accommodates', 'bedrooms']].head(20) # verificar cuál es el patrón que sigue el número de personas y número de habitaciones

accommodates  bedrooms
0              3       1.0
1              7       3.0
2              5       1.0
3              4       2.0
4              2       0.0
5              2       1.0
6              3       1.0
7              2       1.0
8              2       1.0
9              2       1.0
10             2       1.0
11             4       2.0
12             2       1.0
13             6       2.0
14             2       1.0
15             2       1.0
16             2       1.0
17             2       1.0
18             2       1.0
19             8       3.0

In [9]:
df_airbnb['amenities'][0]

'{"Wireless Internet","Air conditioning",Kitchen,Heating,"Family/kid friendly",Essentials,"Hair dryer",Iron,"translation missing: en.hosting_amenity_50"}'

In [10]:
df_airbnb['review_scores_rating'].isna().sum() # total de valores nulos en la califacion

16722

In [11]:
df_airbnb[df_airbnb['review_scores_rating'].isna()]['number_of_reviews'].value_counts() # conteo de propiedades con calificación nula y número de reviews

number_of_reviews
0      15819
1        746
2         86
3         26
4         10
5          5
8          4
15         3
28         2
23         2
10         1
7          1
25         1
63         1
101        1
9          1
17         1
18         1
43         1
47         1
19         1
21         1
131        1
13         1
33         1
32         1
36         1
6          1
37         1
Name: count, dtype: int64

In [12]:
zero_count = (df_airbnb['accommodates'] == 0).sum()
print(f"Cantidad de valores cero en 'accommodates': {zero_count}") # verificar que no haya una propiedad que aloje 0 personas

Cantidad de valores cero en 'accommodates': 0


In [13]:
df_airbnb['property_type'].value_counts() # conteo de tipos de propiedades

property_type
Apartment             49003
House                 16511
Condominium            2658
Townhouse              1692
Loft                   1244
Other                   607
Guesthouse              498
Bed & Breakfast         462
Bungalow                366
Villa                   179
Dorm                    142
Guest suite             123
Camper/RV                94
Timeshare                77
Cabin                    72
In-law                   71
Hostel                   70
Boutique hotel           69
Boat                     65
Serviced apartment       21
Tent                     18
Castle                   13
Vacation home            11
Yurt                      9
Hut                       8
Treehouse                 7
Chalet                    6
Earth House               4
Tipi                      3
Train                     2
Cave                      2
Casa particular           1
Parking Space             1
Lighthouse                1
Island                    1
Name: 

In [14]:
df_airbnb['room_type'].value_counts() # conteo de tipos de habitaciones

room_type
Entire home/apt    41310
Private room       30638
Shared room         2163
Name: count, dtype: int64

In [15]:
df_airbnb['amenities'].value_counts() # conteo de amenidades

amenities
{}                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               586
{"translation missing: en.hosting_amenity_49","translation missing: en.hosting_amenity_50"}                                                                                                                                                                                                                                                                                                                              

In [16]:
df_airbnb['city'].value_counts() # conteo de ciudades

city
NYC        32349
LA         22453
SF          6434
DC          5688
Chicago     3719
Boston      3468
Name: count, dtype: int64